### PR TITLE
map talkable objects to struct

### PR DESCRIPTION
**Pull request by vitalikdanchenko**

### CODE REVIEW

File: lib/talkable/resources/origin.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      if order_date
+        origin_hash[:order_date] = begin
+          Time.iso8601(order_date)
+        rescue ArgumentError
+          order_date
+        end
+      end
+
+      origin = self.new(origin_hash)
+      origin.offer ||= Talkable::Offer.parse(result_hash)
```
Comment (Tonkonozhenko) [inline lib/talkable/resources/origin.rb:18]: Can we use just `=` here?

File: lib/talkable/resources/offer.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module Talkable
+  class Offer < Hashie::Mash
+    def self.parse(result_hash)
+      offer = self.new(result_hash[:offer])
+      offer.claim_links ||= Hashie::Mash.new(result_hash[:claim_links])
```
Comment (Tonkonozhenko) [inline lib/talkable/resources/offer.rb:5]: Can we use just `=` here?

File: lib/talkable/resources/origin.rb (hunk 1 above)
Comment (vitalikdanchenko) [inline lib/talkable/resources/origin.rb:18]: yes, but I suppose API response could have `offer` inside `origin` someday

Comment (Tonkonozhenko) [inline lib/talkable/resources/origin.rb:18]: Ok, now it does not matter

**Comment by Tonkonozhenko** [comment]
Why don't you use this classes in your API calls?

**Comment by vitalikdanchenko** [comment]
I would prefer that API module works only as network wrapper and there is other pull request coming soon that uses this classes

**Comment by Tonkonozhenko** [comment]
I think that `Talkable::API::Origin.create(...)` should return instance of `Talkable::Origin`

**Comment by vitalikdanchenko** [comment]
I'm not sure yet and as I said there will be other pull request exactly with integration functionality

**Comment by Tonkonozhenko** [comment]
LGTM

File: lib/talkable/resources/origin.rb (hunk 1 above)
Comment (bogdan) [inline lib/talkable/resources/origin.rb:18]: I like this idea.... maybe we should even do it now... as offer is already made during purchase generation

**Pull request merged**